### PR TITLE
fixed issues with `AwaitConditionAsync`

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Changes

Fixing API issues reported in https://github.com/akkadotnet/akka.net/issues/6466

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Design discussion issue #6466
* [x] Changes in public API reviewed, if any.

### CODE REVIEW

File: src/core/Akka.TestKit/EventFilter/IEventFilterApplier.cs [hunk 1]
```diff
@@ -123,6 +127,9 @@ public interface IEventFilterApplier
         /// <param name="action">The action.</param>
         /// <param name="cancellationToken"></param>
         Task ExpectAsync(int expectedCount, TimeSpan timeout, Func<Task> action, CancellationToken cancellationToken = default);
+        
+        [Obsolete("Use ExpectAsync<T>(expectedCount, TimeSpan, Func<Task<T>>) instead. This method only exists to support backwards compatibility as of Akka.NET v1.5.")]
```
Comment (Aaronontheweb) [inline src/core/Akka.TestKit/EventFilter/IEventFilterApplier.cs:131]: Added `Obsolete` APIs for `__Async` methods that were introduced in the v1.4 TestKit that accepted `Action` overloads.

File: src/core/Akka.TestKit/TestKitBase_Expect.cs [hunk 2]
```diff
@@ -568,6 +568,14 @@ public async ValueTask<T> ExpectMsgAnyOfAsync<T>(IEnumerable<T> messages, Cancel
 
             return (T)o;
         }
+        
+        public IReadOnlyCollection<T> ExpectMsgAllOf<T>(
```
Comment (Aaronontheweb) [inline src/core/Akka.TestKit/TestKitBase_Expect.cs:572]: Needed to add this for backwards compat.

File: src/core/Akka.TestKit/TestKitBase_Expect.cs [hunk 3]
```diff
@@ -606,6 +614,13 @@ await foreach (var item in enumerable)
                 yield return item;
             }
         }
+        
+        public IReadOnlyCollection<T> ExpectMsgAllOf<T>(
```
Comment (Aaronontheweb) [inline src/core/Akka.TestKit/TestKitBase_Expect.cs:618]: Backwards compat.

**Review by Aaronontheweb — COMMENTED** [review]
Reviewed my changes

**Review by Arkatufus — APPROVED** [review]
LGTM

**Pull request merged**